cmd/wick/helpers.go

```diff
@@ -125,22 +129,317 @@ func readFromProfile(profile string) (*core.ClientInfo, error) {
+func validatePrivateKey(s string) error {
+	if len(s) != 64 && len(s) != 32 {
+		return fmt.Errorf("invalid private key: private key must have length of 32 or 64")
+	}
+	return nil
+}
+
+// getInputFromUser ask user for input if not present in clientInfo.
+func getInputFromUser(serializer string, clientInfo *core.ClientInfo) (*core.ClientInfo, string, error) {
+	var writer = os.Stdout
+	var reader = os.Stdin
+	if clientInfo.Url == "" || clientInfo.Url == "ws://localhost:8080/ws" {
+		inputUrl, err := askForInput(reader, writer, "Enter url", "ws://localhost:8080/ws",
+			true, true, validateURL)
+		if err != nil {
+			return nil, "", err
+		}
+		clientInfo.Url = inputUrl
+	}
+
+	if clientInfo.Realm == "" || clientInfo.Realm == "realm1" {
+		inputRealm, err := askForInput(reader, writer, "Enter realm", "realm1", true,
+			true, validateRealm)
+		if err != nil {
+			return nil, "", err
+		}
+		clientInfo.Realm = inputRealm
+	}
+
+	var serializerStr string
+	if serializer == "json" {
+		inputSerializer, err := askForInput(reader, writer, "Enter serializer", "json", true,
+			true, validateSerializer)
+		if err != nil {
+			return nil, serializerStr, err
+		}
+		serializerStr = inputSerializer
+	}
+
+	if clientInfo.Authid == "" {
+		inputAuthid, err := askForInput(reader, writer, "Enter authid", "", false,
+			false, nil)
+		if err != nil {
+			return nil, serializerStr, err
+		}
+		clientInfo.Authid = inputAuthid
+	}
+
+	if clientInfo.Authrole == "" {
+		inputAuthrole, err := askForInput(reader, writer, "Enter authrole", "", false,
+			false, nil)
+		if err != nil {
+			return nil, serializerStr, err
+		}
+		clientInfo.Authrole = inputAuthrole
+	}
+
+	if clientInfo.AuthMethod == "" || clientInfo.AuthMethod == "anonymous" {
+		inputAuthMethod, err := askForInput(reader, writer, "Enter authmethod", "anonymous", true,
+			true, validateAuthMethod)
+		clientInfo.AuthMethod = inputAuthMethod
+		if err != nil {
+			return nil, serializerStr, err
+		}
+	}
+
+	switch clientInfo.AuthMethod {
+	case ticketAuth:
+		inputTicket, err := askForInput(reader, writer, "Enter ticket", "", true,
+			true, nil)
+		if err != nil {
+			return nil, serializerStr, err
+		}
+		clientInfo.Ticket = inputTicket
+	case wampCraAuth:
+		inputSecret, err := askForInput(reader, writer, "Enter secret", "", true,
+			true, nil)
+		if err != nil {
+			return nil, serializerStr, err
+		}
+		clientInfo.Secret = inputSecret
+	case cryptosignAuth:
+		inputPrivateKey, err := askForInput(reader, writer, "Enter private key", "", true,
+			true, validatePrivateKey)
+		if err != nil {
+			return nil, serializerStr, err
+		}
+		clientInfo.PrivateKey = inputPrivateKey
+	}
+
+	return clientInfo, serializerStr, nil
+}
+
+// ensureFile create file and all directories in given path if not already exists.
+func ensureFile(filePath string) error {
+	_, err := os.Stat(filePath)
+	if os.IsNotExist(err) {
+		if err = os.MkdirAll(filepath.Dir(filePath), os.ModePerm); err != nil {
```

Comment:
os.ModePerm is a mask for obtaining the permission bits from stat's file mode info. So in this case, the code creates a directory with permissions 0777 (read & write by all) which is wrong.

```suggestion
		if err = os.MkdirAll(filepath.Dir(filePath), 0755); err != nil {
```

Reply:
Done